File: client/dto/OrderDTO.js
class OrderDTO {
    constructor(order_id, date, total, payment_method, points, customer_id, employee_id, branch_id, shoes, accessories) {
        this.order_id = order_id;
        this.date = date;
        this.total = total;
        this.payment_method = payment_method;
        this.points = points;
        this.customer_id = customer_id;
        this.employee_id = employee_id;
        this.branch_id = branch_id;
        this.shoes = shoes;
        this.accessories = accessories;
    }
}

File: server/src/main/java/software/kasunkavinda/dto/OrderDTO.java
package software.kasunkavinda.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import software.kasunkavinda.enums.Payment_Method;

import java.util.Date;
import java.util.List;
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder

public class OrderDTO {

    private String order_id;

    @NotNull(message = "Order date cannot be null")
    private Date date;

    @NotNull(message = "Total amount cannot be null")
    private Double total;

    @NotNull(message = "Payment method cannot be null")
    private Payment_Method payment_method;

    @NotNull(message = "Points cannot be null")
    private Integer points;

    @NotEmpty(message = "Customer ID cannot be empty")
    private String customer_id;

    @NotEmpty(message = "Employee ID cannot be empty")
    private String employee_id;

    @NotEmpty(message = "Branch ID cannot be empty")
    private String branch_id;

    private String refund_id;

    private List<ShoeDTO> shoes;

    private List<AccessoriesDTO> accessories;

}
